src/Hintman/Hint/NoNewlineAtFileEnd.hs

```diff
@@ -0,0 +1,84 @@
+module Hintman.Hint.NoNewlineAtFileEnd
+       ( getNoNewlineAtFileEndComments
+       ) where
+
+import Hintman.Core.Hint (Hint, HintType (NoNewlineAtFileEnd), Line (..), simpleSuggestion, toLines)
+import Hintman.Core.PrInfo (ModifiedFile (..))
+import Hintman.Core.Review (Comment (..), createComment)
+
+import qualified Data.Text as T
+
+
+-- | Creates 'Comment's on adding newlines in all modified files.
+getNoNewlineAtFileEndComments :: [ModifiedFile] -> [Comment]
+getNoNewlineAtFileEndComments = mapMaybe getFileNoNewlineAtFileEndComment
+
+{- | Takes a file and returns a comment if file doesn't end with a '\n'.
+The suggestion text for the line in comment is the context of the line with '\n' appended. -}
+getFileNoNewlineAtFileEndComment :: ModifiedFile -> Maybe Comment
+getFileNoNewlineAtFileEndComment mf@ModifiedFile{..} = spawnComment $
+    maybeGetNonEmptyLastLine fileLines diffText
+  where
+    diffText :: Text
+    diffText = decodeUtf8 mfContent
+
+    fileLines :: [Line]
+    fileLines = toLines diffText
+
+    {- | Spawns a "NoNewlineAtFileEnd" comment on a given line, suggesting the same
+    line, but with '\n' appended. -}
+    spawnComment :: Maybe Line -> Maybe Comment
+    spawnComment = (=<<) (\line -> createComment mf line $ createHint line)
+
+    createHint :: Line -> Hint
+    createHint line = simpleSuggestion NoNewlineAtFileEnd "No newline at end of file" $ suggestionText line
+
+    suggestionText :: Line -> Text
+    suggestionText Line{..} = appendNewline lineBody
+
+    appendNewline :: Text -> Text
+    appendNewline line = T.append line "\n"
+
+{- | Checks if file ends with a newline, and if it does not, returns the last line (to spawn warning on).
+
+More precise explanation of the logic within function:
+
+1. We check the last symbol of file (being unaware of Lines).
+Since file can be empty, the return type is 'Maybe Char'.
+Then, we split file into lines (since Line is an essential concept of spawning a lint).
+Here an important note is that result of lines will trim a '\n' symbol from the last line, so
+we can't just check the last symbol of line, we have to check last symbol of raw content).
```

Comment:
I guess this description is different from what happens below in the code. You first get the last line instead of the last character (it's even commented so).

I would suggest first check for is the file is empty. If it is then stop. If it's not then check the last symbol. If it's not a newline then we need to start looking for the last line number to create the suggestion, but only in that case.

Reply:
It makes sense, will implement in a few minutes :)

Reply:
Oh, I rebased the branch and forgot to use `--force` on push,  =_=
Now it is [implemented](https://github.com/kowainik/hintman/pull/156/files#diff-4a26a42e1f8356b6a5580807d6b2a986R67), we only split if the last symbol is not `\n`.